MantidPlot/src/FunctionCurve.cpp

```diff
@@ -271,36 +256,23 @@ void FunctionCurve::loadMantidData(Mantid::API::MatrixWorkspace_const_sptr ws,
     if (wi >= ws->getNumberHistograms())
       return;
 
-    const Mantid::MantidVec &wsX = ws->readX(wi);
+    const auto &wsXPoints = ws->points(wi);
 
-    if (d_from < wsX.front()) {
-      d_from = wsX.front();
-    }
-    if (d_to > wsX.back()) {
-      d_to = wsX.back();
-    }
+    if (d_from < wsXPoints.front())
+      d_from = wsXPoints.front();
+    if (d_to > wsXPoints.back())
+      d_to = wsXPoints.back();
 
     std::vector<double> X;
-    X.reserve(wsX.size());
-
-    if (ws->isHistogramData()) {
-      for (size_t i = 0; i < ws->blocksize() - 1; i++) {
-        double x = (wsX[i] + wsX[i + 1]) / 2;
-        if (x < d_from)
-          continue;
-        if (x > d_to)
-          break;
-        X.push_back(x);
-      }
-    } else {
-      for (size_t i = 0; i < ws->blocksize(); i++) {
-        double x = wsX[i];
-        if (x < d_from)
-          continue;
-        if (x > d_to)
-          break;
-        X.push_back(x);
-      }
+    X.reserve(wsXPoints.size());
+
+    for (int i = 0; i < static_cast<int>(ws->blocksize()); i++) {
+      double x = wsXPoints[i];
+      if (x < d_from)
+        continue;
+      if (x > d_to)
+        break;
+      X.push_back(x);
```

Comment:
While it is duplicated, I'm not sure this is really important enough to need its own function - it does little more than copy x-point data to a vector in a specific manner. Doesn't seem worth it to me.